test/testdata/dsl/default_args.rb.dsl-tree.exp

```diff
@@ -0,0 +1,13 @@
+class <emptyTree><<C <root>>> < ()
+  def foo<<C <todo sym>>>(a = <emptyTree>, b = <emptyTree>, &<blk>)
+    <emptyTree>
+  end
+
+  def foo<defaultAarg>1<<C <todo sym>>>(a = <emptyTree>, b = <emptyTree>, &<blk>)
```

Comment:
Why does the default arg function get the values of `a` and `b` as arguments? Since this is the first arg, is it allowed to depend on the value of `b`?

Reply:
I was trying to keep the function signature to exactly match up so that I can clone the `sig` easier and so that the compiler can match the ABI easier. The parser will actually take care of the care you are talking about by rewriting the local variable access to be `self.b` and will throw a NoMethod error.